openstackmap.py

```diff
@@ -24,245 +24,480 @@
 #
 author = 'chema'
 
-from osclients import osclients
+from osclients import osclients, OpenStackClients
+
 import cPickle as pickle
 import os
+import time
+from os import environ as env
+
+class OpenStackMap():
+    """
+    This class build a map from the resources (VMs, networks, images,
+    volumes, users, tenants, roles...) in an OpenStack infrastructure.
+
+    This map is optionally cached on disk and can be accessed offline or can be
+    used as a metadata snapshot of an OpenStack infrastructure (e.g. to
+    analyse resource use and build statistics)
+    """
+
+    # objects strategy see the __init__ method documentation
+    DIRECT_OBJECTS, NO_CACHE_OBJECTS, REFRESH_OBJECTS, USE_CACHE_OBJECTS = \
+        range(4)
+
+    # If use_wrapper is True, dictionaries are wrapped to allow access to
+    # resource['field'] also as resource.field. This is not used when
+    # objects_strategy is DIRECT_OBJECTS
+    use_wrapper = True
+
+    def __init__(
+            self, persistence_dir='~/persistence', region=None, auth_url=None,
```

Comment:
This is only a default value, but you're right. I'll change ~/persistence to ~/openstackmap